lib/rules/media-feature-name-no-unknown/__tests__/index.js

```diff
@@ -140,18 +152,12 @@ testRule({
 		{
 			code: '@media @smartphones { }',
 		},
-	],
-
-	reject: [
 		{
 			code: '@media (unknown: @tablet) { }',
-			message: messages.rejected('unknown'),
-			line: 1,
-			column: 9,
-			endLine: 1,
-			endColumn: 16,
```

Comment:
🤔 Is there a good reference for the kinds of syntax that should be allowed?
If everything less/scss related is still a valid sequence of tokens and blocks then I think we can add support for a "non-strict" mode.

I will look into this.

Reply:
> Is there a good reference for the kinds of syntax that should be allowed?

No. We currently have allowed only standard syntax:

https://github.com/stylelint/stylelint/blob/0c328f919fe7f2588baa6f6907a5520dac6fe6c4/docs/developer-guide/rules.md?plain=1#L11-L13

But for historical reasons, some rules can process non-standard syntaxes. I'm worried that this PR can be a breaking change for non-standard syntaxes users, and so they may complain through reporting issues. 😓 

That's why I asked the question above. I would be glad if `@csstools/media-query-list-parser` could easily provide a "non-strict" mode.

Reply:
I meant for less/scss in general, something I can use in tests for `@csstools/media-query-list-parser`.

With a good list of examples I should be able to add a "non-strict" mode.
Issue mainly is that I don't use scss/less, so I don't have a good sense of what I should expect here.

Reply:
```scss
$breakpoint-tablet: 768px;
@media (min-width: $breakpoint-tablet) {
  .foo { color: green; }
}

$breakpoint-feature: width;
@media (min-#{$breakpoint-feature}: 100px) {
  .foo { color: green; }
}

$breakpoint-value-part: 10;
@media (#{$breakpoint-value-part * 2}px < width < 100px) {
  .foo { color: green; }
}
```

I don't think a non-strict mode is an option.

The parser needs to find a specific sequence of component values to determine what kind of node something.

In `scss` `#{$breakpoint-value-part * 2}px` is a single thing that will become `20px` but in CSS that is a delim `#`, followed by a block `{...}`, followed by an ident `px`.

------

Going to see what we can do with `GeneralEnclosed`, that is the fallback type for everything non-standard.

Reply:
@ybiquitous I've updated this proposal with a very basic parser algorithm that should be able to extract standard CSS media feature names from non standard conditions.

Because we are not building an AST and aren't trying to fit this data into a typed object model, we can take a lot of shortcuts and still extract the feature name.

Reply:
Thanks for your effort! Looks good 👍🏼